adapters/intertech/intertech.go

```diff
@@ -0,0 +1,326 @@
+package intertech
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"text/template"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/adapters"
+	"github.com/prebid/prebid-server/v3/config"
+	"github.com/prebid/prebid-server/v3/errortypes"
+	"github.com/prebid/prebid-server/v3/macros"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+)
+
+const (
+	refererQueryKey  = "target-ref"
+	currencyQueryKey = "ssp-cur"
+	impIdQueryKey    = "imp-id"
+)
+
+// intertechPlacementID is the composite id of an ad placement
+type intertechPlacementID struct {
+	PageID string
+	ImpID  string
+}
+
+type adapter struct {
+	endpoint *template.Template
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	template, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+
+	bidder := &adapter{
+		endpoint: template,
+	}
+
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(requestData *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var (
+		requests []*adapters.RequestData
+		errors   []error
+	)
+
+	referer := getReferer(requestData)
+	currency := getCurrency(requestData)
+
+	for i := range requestData.Imp {
+		imp := requestData.Imp[i]
+
+		placementId, err := getPlacementID(imp)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		if err := modifyImp(&imp); err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		resolvedUrl, err := a.resolveUrl(*placementId, referer, currency)
+		if err != nil {
+			errors = append(errors, err)
+			continue
+		}
+
+		splittedRequestData := splitRequestDataByImp(requestData, imp)
+
+		requestBody, err := json.Marshal(splittedRequestData)
```

Comment:
Can you please switch this to using jsonutil? Starting in PBS 3.0 we're switching to our more optimized implementation. Thanks.